src/Security/LogOutAuthenticationHandler.php

```diff
@@ -58,18 +70,60 @@ public function logOut(HTTPRequest $request = null)
         $loginHandler = Injector::inst()->get(LogInAuthenticationHandler::class);
         $member = Security::getCurrentUser();
 
-        if (RememberLoginHash::config()->get('logout_across_devices')) {
-            foreach ($member->LoginSessions() as $session) {
-                $session->delete();
+        // Members without admin access are unable to access session manager and individually revoke login sessions
+        // These types of members often exist on sites where website users can create accounts via the frontend
+        // and a corresponding record is created on the Member table
+        // Since there's no other way for these users to logout any malicious devices, auto log out of all devices
+        // when one device is logged out
+        $noAdminAccessRevokeAll = static::config()->get('no_admin_access_revoke_all_on_logout');
+        if ($noAdminAccessRevokeAll && !$this->hasAdminAccess($member)) {
+            foreach ($member->LoginSessions() as $loginSession) {
+                $loginSession->delete();
             }
         } else {
             $loginSessionID = $request->getSession()->get($loginHandler->getSessionVariable());
             $loginSession = LoginSession::get()->byID($loginSessionID);
-            if ($loginSession && $loginSession->canDelete($member)) {
+            if (
+                $loginSession && $loginSession->canDelete($member) ||
+                $loginSession && !$noAdminAccessRevokeAll && !$this->hasAdminAccess($member)
+            ) {
                 $loginSession->delete();
             }
         }
 
         $request->getSession()->clear($loginHandler->getSessionVariable());
     }
+
+    /**
+     * Decides whether the provided user has access to any LeftAndMain controller, which indicates some level
+     * of access to the CMS.
+     *
+     * @see LeftAndMain::init()
+     * @param Member $member
+     * @return bool
+     */
+    private function hasAdminAccess(Member $member): bool
```

Comment:
This function was copy pasted from https://github.com/silverstripe/silverstripe-mfa/blob/4/src/Service/EnforcementManager.php#L221

It's not pretty, but it works

It's a `private` function so easy to remove in the future if we can a function like this to core